server/src/main/java/org/opensearch/index/remote/RemoteTranslogTransferTracker.java

```diff
@@ -0,0 +1,734 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.remote;
+
+import org.opensearch.common.util.MovingAverage;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.index.shard.ShardId;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Stores Remote Translog Store-related stats for a given IndexShard.
+ *
+ * @opensearch.internal
+ */
+public class RemoteTranslogTransferTracker {
+    /**
+     * The shard that this tracker is associated with
+     */
+    public final ShardId shardId;
+
+    /**
+     * Epoch timestamp of the last successful Remote Translog Store upload.
+     */
+    private final AtomicLong lastSuccessfulUploadTimestamp;
```

Comment:
volatile should suffice the need here. AtomicLong has introduces more performance overhead compared to volatile. In this case only one remote translog upload can happen at a time, we can use volatile.

Reply:
Pls ensure the above for other AtomicLong fields here.

Reply:
Are we planning concurrent uploads of multiple segment/tlog files in the future (not talking about multipart upload)? If not, it makes sense to convert these to `volatile`.

Reply:
@ashking94 based on our discussion earlier, I am keeping the `AtomicLong` fields as is because the uploads are concurrent. Please re-open if you have any other recommendations for this.